src/Handler/FilesystemHandler.php

```diff
@@ -28,14 +28,25 @@ class FilesystemHandler extends \SessionHandler implements HandlerInterface
 	 */
 	public function __construct(string $path = '')
 	{
-		$path = $path ?: ini_get('session.save_path');
+		$pathConfig = ini_get('session.save_path');
 
-		// If the path is still empty, then we can't use this handler
-		if (empty($path))
+		// If the paths are empty, then we can't use this handler
+		if (empty($path) && empty($pathConfig))
 		{
 			throw new \InvalidArgumentException('Invalid argument $path');
 		}
 
+		// If path is empty or equal to the the PHP configured path, set only the handler and use the PHP path directly
+		if (empty($path) || $path === $pathConfig)
```

Comment:
We should check the directory exists and that it is writeable. On all the shared hosts that are well configured this should be fine and then we can do this. On hosts where this isn't the case we can then try and create it (and if that fails so be it)

Reply:
Something like https://gist.github.com/juanramon/787624#file-session-php-L2 if we are going to stop here

Reply:
This will not work! You cannot write or check this folder if the directory is outside the open_basedir paths.

Result from the code snippet you've linked:

```
Warning: file_exists(): open_basedir restriction in effect. File(/var/lib/php/session) is not within the allowed path(s): (/var/www/vhosts/code-sprint.de/:/tmp/) in /var/www/vhosts/code-sprint.de/joomla4.code-sprint.de/libraries/vendor/joomla/session/src/Handler/FilesystemHandler.php on line 32
Session path doesn't exist: /var/lib/php/session
```

It is the task of PHP (internally) to manage the sessions correctly in this case.

Reply:
That's sad :) oh well